### DISCUSSION TITLE
Does Serverpod supports JWT token implementation for authentication?

### DISCUSSION THREAD

**Discussion by Shubhamdsingh** [serverpod/serverpod#524] [Q&A] [answered]
Does Serverpod supports JWT token implementation for authentication?
 if yes is there any dedicated package for it.?
if no does serverpod support any token based authentication ?

**Comment by yahu1031** [comment]
@Shubhamdsingh Please don't tag us. We already subscribed to any actions on the repo so we get notifications by default. I'm not aware of it. But AFAIS nope. Use `jaguar_jwt` to create JWT and use it. We use the ID in the table as a unique id.

**Reply by Shubhamdsingh** [reply]
Ok thanks

**Comment by vlidholt** [comment, marked as answer]
If you are talking authentication with e.g. Apple, that uses JWT. For it's own authentication, Serverpod stores a key on the user device to keep a secure session open. Look at the `serverpod_auth` module for more information.

Also, the documentation is here: https://docs.serverpod.dev/concepts/authentication
